Data/models/Domain.swift

```diff
@@ -200,31 +200,34 @@ extension Domain {
     return domainOnCorrectContext
   }
 
-  class func deleteNonBookmarkedAndClearSiteVisits(
-    context: NSManagedObjectContext,
-    _ completionOnMain: @escaping () -> Void
-  ) {
-
-    let fetchRequest = NSFetchRequest<Domain>()
-    fetchRequest.entity = Domain.entity(context)
-    do {
-      let results = try context.fetch(fetchRequest)
-      results.forEach {
-        if let bms = $0.bookmarks, bms.count > 0 {
-          // Clear visit count
-          $0.visits = 0
-        } else {
-          // Delete
-          context.delete($0)
+  public class func deleteNonBookmarkedAndClearSiteVisits(_ completionOnMain: @escaping () -> Void) {
+    DataController.perform { context in
+      let fetchRequest = NSFetchRequest<Domain>()
+      fetchRequest.entity = Domain.entity(context)
+      do {
+        let results = try context.fetch(fetchRequest)
+        results.forEach {
+          if let bms = $0.bookmarks, bms.count > 0 {
+            // Clear visit count and clear the shield settings
+            $0.visits = 0
+            $0.shield_allOff = nil
+            $0.shield_adblockAndTp = nil
+            $0.shield_noScript = nil
+            $0.shield_fpProtection = nil
+            $0.shield_safeBrowsing = nil
```

Comment:
I don't think needed in this PR, but I wonder if we should reset `wallet_permittedAccounts`? If we can do it in wallet panel branch cc @kylehickinson